contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -262,6 +262,17 @@ def get_channel(self, short_channel_id: Union[ShortChannelId, str]):
             short_channel_id = ShortChannelId.from_str(short_channel_id)
         return self.channels.get(short_channel_id)
 
+    def get_halfchannel(self,
+                        short_channel_id: Union[ShortChannelId, str],
+                        direction: int):
+        """ Returns a GossmapHalfchannel identified by a scid and direction. """
+        assert short_channel_id is not None
+        if isinstance(short_channel_id, str):
+            short_channel_id = ShortChannelId.from_str(short_channel_id)
+        assert direction in [0, 1], "direction can only be 0 or 1"
+        channel = self.get_channel(short_channel_id)
+        return channel.half_channels[direction]
```

Comment:
Can this result in an `IndexError`? If yes, we should likely use `channel.half_channel.get(direction, None)` and then check for `None` and raise an error if it is `None`.

Reply:
Hm, yes and no. `half_channels` is initialized with `self.half_channels: List[Optional[GossmapHalfchannel]] = [None, None]` so it wouldn't run into index error unless a halfchannel is deliberately deleted using `del()`.

I think having it initialized with `[None, None]` is actually smart as it reduces the amount of checks the code has to do (as you suggested).